htdocs/api/class/api_documents.class.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/* Copyright (C) 2016   Xebax Christy           <xebax@wanadoo.fr>
+ * Copyright (C) 2016	Laurent Destailleur		<eldy@users.sourceforge.net>
+ * Copyright (C) 2016   Jean-François Ferry     <jfefe@aternatik.fr>
+ *
+ * This program is free software you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use Luracast\Restler\RestException;
+use Luracast\Restler\Format\UploadFormat;
+
+
+require_once DOL_DOCUMENT_ROOT.'/main.inc.php';
+
+/**
+ * API class for receive files
+ *
+ * @access protected
+ * @class Documents {@requires user,external}
+ */
+class Documents extends DolibarrApi
+{
+
+    /**
+     * @var array   $DOCUMENT_FIELDS     Mandatory fields, checked when create and update object
+     */
+    static $DOCUMENT_FIELDS = array(
+        'name',
+        'modulepart',
+        'file'
+    );
+
+    /**
+     * Constructor
+     */
+    function __construct()
+    {
+        global $db;
+        $this->db = $db;
+    }
+
+    /**
+     * Return a document
+     *
+     * @param   string  $module_part    Module part for file
+     * @param   string  $filename       File name
+     *
+     * @return array
+     * @throws RestException
+     *
+     */
+     public function get($module_part, $filename) {
+
+     }
+
+
+    /**
+     * Receive file
+     *
+     * @param   array   $request_data   Request datas
+     *
+     * @return  bool     State of copy
+     * @throws RestException
+     */
+    public function post($request_data) {
+        global $conf;
+        
+        require_once DOL_DOCUMENT_ROOT . '/core/lib/files.lib.php';
+
+        if (!DolibarrApiAccess::$user->rights->ecm->upload) {
```

Comment:
To fix security, we must enhance this to use
dol_check_secure_access_document( )
to check if user has permission depending on "modulepart" value.

Problem: this method currently can be used to test user has permission to read, but not permission to write. We must also add a parameter to test on ->write instead of ->lire